### PR TITLE
Add tests for API  categories feature

### PR DESCRIPTION
**Pull request by msm1992**
This PR adds tests for API Categories feature.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                String responsePayload = EntityUtils.toString(entity);
+                JSONParser parser = new JSONParser();
+                JSONObject object = (JSONObject) parser.parse(responsePayload);
+                int count = (int) (long) object.get("count");
+                Assert.assertEquals(count, 1);
+            }
+        }
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test delete API category")
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java]: Let's add depends on "testGetAPICategoriesFromAdminAPI"

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            put.setEntity(payload);
+            try (CloseableHttpResponse response = client.execute(put);) {
+                HttpEntity entity = response.getEntity();
+                Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+            }
+        }
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Test delete API category",
+            dependsOnMethods = { "testAddAPICategory" })
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java]: Let's change this depends on to "testUpdateAPICategory"

Comment (msm1992) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java]: fuxed in https://github.com/wso2/product-apim/pull/7245/commits/01de486f9a550d1f94acf9bb03fd024128f46e3f

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java (hunk 1 above)
Comment (msm1992) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APICategoriesTestCase.java]: fixed in https://github.com/wso2/product-apim/pull/7245/commits/01de486f9a550d1f94acf9bb03fd024128f46e3f

**Pull request merged**